libs/localization/scripts/extract.spec.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  convertToLocalArray,
+  convertToContentfulArray,
+  mergeArray,
+} from './extract'
+
+describe('extractFunctions', () => {
+  it('should use local values if none is defined in contentful', () => {
+    const localValues = {
+      'application.system:no.value.in.contentful': {
+        defaultMessage: 'Takk',
+        description: 'Thanks',
+      },
+    }
+
+    const contentfulValues = {
+      fields: {
+        strings: {
+          'is-IS': [
+            {
+              id: 'application.system:no.value.in.contentful',
+              defaultMessage: '',
+              'is-IS': '',
+              en: '',
+            },
+          ],
+        },
+      },
+    }
+
+    const local = convertToLocalArray(localValues)
+    const distant = convertToContentfulArray(contentfulValues as any)
+
+    expect(mergeArray(local, distant)).toStrictEqual([
+      {
+        id: 'application.system:no.value.in.contentful',
+        defaultMessage: 'Takk',
+        'is-IS': 'Takk',
+        en: 'Thanks',
+      },
```

Comment:
We should store `description` in a special field and show it in the Contentful UI, since it's very useful to provide context to translators. 

Also, not sure if we should default English to the description, as it might discourage developers from writing helpful descriptions.

Reply:
Yes, I've been noticing and doing it on my own as well that the `description` is mostly used to define the English translation to the Icelandic string in the `defaultMessage`.

It's just a very convenient/workaround way to define both strings at the same place at once, especially for non-native-Icelandic people to get quickly an interface they can "google translate" and switch to English for development until some proper translations are added through contentful. That's the reason I was using the `description` to populate the English field inside contentful. However, I agree it's not the real purpose of the `description` field.

I'm totally fine to stop doing that and going inside Contentful to define the English strings, but will it be the case of all developers and how can we enforce this so it's not used the wrong way? 

For the parental leave, for example, 90% of the strings were already translated inside the messages.ts, using the `description` field, but because we are not doing using it to populate the `English` fields inside contentful, all the English translation were missing once switching to the English language. I pretty much had to copy/paste everything between messages.ts and contentful to populate the English translation.